enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -97,12 +100,29 @@ secret's contents to files and specify these in the kube-apiserver configuration
 
 #### Backend - Integrated OAuth Server Token Authentication
 
+0. cluster-authentication-operator adds a new `system:openshift:openshift-token-validation`
+   cluster role to its manifests:
+   ```yaml
+   rules:
+   - nonResourceURLs:
+     - "/tokenvalidation"
+     verbs:
+     - "post"
+   ```
 1. cluster-authentication-operator observes `authentication.config/cluster` `type` field
-2. if set to "IntegratedOAuth", it creates an `openshift-config/webhook-authentication-integrated-oauth`
-  secret and adds it to the `webhookTokenAuthenticator` field of the `authentication.config/cluster`
-  resource. If the `type` field is set to any other value, the operator will perform no action to the
-  `webhookTokenAuthenticator` field.
-  - the above secret contains details that point kube-apiserver to the oauth-apiserver for token reviews
+2. if the `type` field is set to "IntegratedOAuth":
+    - it creates an `openshift-oauth-apiserver/openshift-authenticator` service account and ensures that
+      a `system:openshift:openshift-token-validation` clusterrolebinding that binds this SA to the
+      `system:openshift:openshift-token-validation` clusterrole exists
+    - it waits for the service account to be provided with a token from the kube-controller-manager
+    - it creates an `openshift-config/webhook-authentication-integrated-oauth`
+      secret  and adds it to the `webhookTokenAuthenticator.kubeConfig` field of
+      the `authentication.config/cluster` resource
+    - it sets the `webhookTokenAuthenticator.cacheTTL` to 0 to prevent premature token timeout
```

Comment:
I would like to avoid the issue demonstrated in the picture

The default cache TTL as documented at https://github.com/openshift/api/blob/0c4b3ae5f5df4a7c18c76f492a61fa9a39e56d5f/kubecontrolplane/v1/types.go#L159-L162 is 2minutes, whereas the minimum token timeout period is 5 minutes, so in the worst case we might shorten the validity of a token by 40%.

![token_timeout](https://user-images.githubusercontent.com/718169/88059217-d6416700-cb64-11ea-80c1-8061d7ebcd65.png)